common/client/transaction_sender_test.go

```diff
@@ -0,0 +1,360 @@
+			require.EqualError(t, err, "MultiNode has already been stopped: already stopped")
+		}
+		err = txSender.Close()
+		if err != nil {
+			// Allow TransactionSender to be closed early for testing
+			require.EqualError(t, err, "TransactionSender has already been stopped: already stopped")
+		}
+	})
+	return &mn, txSender
+}
+
+func classifySendTxError(_ any, err error) SendTxReturnCode {
+	if err != nil {
+		return Fatal
+	}
+	return Successful
+}
+
+func TestTransactionSender_SendTransaction(t *testing.T) {
+	t.Parallel()
+
+	newNodeWithState := func(t *testing.T, state NodeState, txErr error, sendTxRun func(args mock.Arguments)) *mockNode[types.ID, SendTxRPCClient[any]] {
+		rpc := newSendTxRPC(txErr, sendTxRun)
+		node := newMockNode[types.ID, SendTxRPCClient[any]](t)
+		node.On("String").Return("node name").Maybe()
+		node.On("RPC").Return(rpc).Maybe()
+		node.On("State").Return(state).Maybe()
+		node.On("Close").Return(nil).Once()
+		return node
+	}
+
+	newNode := func(t *testing.T, txErr error, sendTxRun func(args mock.Arguments)) *mockNode[types.ID, SendTxRPCClient[any]] {
+		return newNodeWithState(t, NodeStateAlive, txErr, sendTxRun)
+	}
+
+	t.Run("Fails if there is no nodes available", func(t *testing.T) {
+		lggr, _ := logger.TestObserved(t, zap.DebugLevel)
+		_, txSender := newTestTransactionSender(t, types.RandomID(), lggr, nil, nil)
+		_, err := txSender.SendTransaction(tests.Context(t), nil)
+		assert.EqualError(t, err, ErroringNodeError.Error())
+	})
+
+	t.Run("Transaction failure happy path", func(t *testing.T) {
+		expectedError := errors.New("transaction failed")
+		mainNode := newNode(t, expectedError, nil)
+		lggr, observedLogs := logger.TestObserved(t, zap.DebugLevel)
+
+		_, txSender := newTestTransactionSender(t, types.RandomID(), lggr,
+			[]Node[types.ID, SendTxRPCClient[any]]{mainNode},
+			[]SendOnlyNode[types.ID, SendTxRPCClient[any]]{newNode(t, errors.New("unexpected error"), nil)})
+
+		result, sendErr := txSender.SendTransaction(tests.Context(t), nil)
+		require.ErrorIs(t, sendErr, expectedError)
+		require.Equal(t, Fatal, result)
+		tests.AssertLogCountEventually(t, observedLogs, "Node sent transaction", 2)
+		tests.AssertLogCountEventually(t, observedLogs, "RPC returned error", 2)
+	})
+
+	t.Run("Transaction success happy path", func(t *testing.T) {
+		mainNode := newNode(t, nil, nil)
+
+		lggr, observedLogs := logger.TestObserved(t, zap.DebugLevel)
+		_, txSender := newTestTransactionSender(t, types.RandomID(), lggr,
+			[]Node[types.ID, SendTxRPCClient[any]]{mainNode},
+			[]SendOnlyNode[types.ID, SendTxRPCClient[any]]{newNode(t, errors.New("unexpected error"), nil)})
+
+		result, sendErr := txSender.SendTransaction(tests.Context(t), nil)
+		require.NoError(t, sendErr)
+		require.Equal(t, Successful, result)
+		tests.AssertLogCountEventually(t, observedLogs, "Node sent transaction", 2)
+		tests.AssertLogCountEventually(t, observedLogs, "RPC returned error", 1)
```

Comment:
Looks like there was only 1? https://github.com/smartcontractkit/chainlink/blob/2521c4b3dee70a9b6210210dd4aa0cb3434a8838/common/client/multi_node_test.go#L721

There should only be a single error since the sendOnlyNode throws an error while the primary node does not. The return code should be success since we don't care about the sendOnlyNodes for their responces.